backtrace-library/src/main/java/backtraceio/library/metrics/BacktraceMetrics.java

```diff
@@ -0,0 +1,339 @@
+    }
+
+    protected String getStartupUniqueEventName() {
+        return this.startupUniqueEventName;
+    }
+
+    public void setStartupUniqueEventName(String startupUniqueEventName) {
+        this.startupUniqueEventName = startupUniqueEventName;
+    }
+
+    public String getBaseUrl() {
+        return this.settings.getBaseUrl();
+    }
+
+    /**
+     * Send startup event to Backtrace
+     */
+    public void sendStartupEvent() {
+        uniqueEventsHandler.sendStartupEvent(startupUniqueEventName);
+        summedEventsHandler.sendStartupEvent(startupSummedEventName);
+    }
+
+    /**
+     * Send all outgoing messages (unique and summed) currently queued
+     */
+    public void send() {
+        uniqueEventsHandler.send();
+        summedEventsHandler.send();
+    }
+
+    /**
+     * Add a unique event to the next Backtrace Metrics request
+     *
+     * @param attributeName Attribute name
+     * @return true if success
+     */
+    public boolean addUniqueEvent(String attributeName) {
+        return addUniqueEvent(attributeName, null);
+    }
+
+    /**
+     * Add a unique event to the next Backtrace Metrics request
+     *
+     * @param attributeName Attribute name
+     * @param attributes    Event attributes
+     * @return true if success
+     */
+    public boolean addUniqueEvent(String attributeName, Map<String, Object> attributes) {
+        if (!shouldProcessEvent(attributeName)) {
+            BacktraceLogger.w(LOG_TAG, "Skipping report: Reached store limit or event has empty name.");
+            return false;
+        }
+
+        Map<String, Object> localAttributes = new HashMap<String, Object>();
+
+        if (attributes != null) {
+            localAttributes.putAll(attributes);
+        }
+
+        BacktraceAttributes backtraceAttributes = new BacktraceAttributes(context, null, customReportAttributes);
+        localAttributes.putAll(backtraceAttributes.getAllBacktraceAttributes());
+
+        DeviceAttributesHelper deviceAttributesHelper = new DeviceAttributesHelper(context);
+        localAttributes.putAll(deviceAttributesHelper.getDeviceAttributes());
+
+        // validate if unique event attribute is available and
+        // prevent undefined attributes
+        Object value = localAttributes.get(attributeName);
+        if (!BacktraceStringHelper.isObjectValidString(value)) {
+            BacktraceLogger.w(LOG_TAG, "Attribute name for Unique Event is not available in attribute scope");
+            return false;
+        }
+        // skip already defined unique events
+        for (UniqueEvent uniqueEvent : uniqueEventsHandler.events) {
+            if (uniqueEvent.getName().equals(attributeName)) {
+                BacktraceLogger.w(LOG_TAG, "Already defined unique event with this attribute name, skipping");
+                return false;
+            }
+        }
+
+        UniqueEvent uniqueEvent = new UniqueEvent(attributeName, System.currentTimeMillis() / 1000, localAttributes);
```

Comment:
In 4 places you are using `System.currentTimeMillis() / 1000` maybe it is worth to extract it to at least static method

Reply:
Good idea, thank you!